diffpy/snmf/subroutines.py

```diff
@@ -1,9 +1,33 @@
 import numpy as np
 from diffpy.snmf.optimizers import get_weights
 from diffpy.snmf.factorizers import lsqnonneg
+from diffpy.snmf.containers import ComponentSignal
 import numdifftools
 
 
+def create_components(number_of_components, number_of_signals, grid_vector):
```

Comment:
Let's call it `initialize_components` so it is clear that at the end they are all empty, but ready to go.